test/tools/yulopti.cpp

```diff
@@ -91,33 +92,32 @@ class YulOpti
 	{
 		ErrorList errors;
 		ErrorReporter errorReporter(errors);
-		CharStream _charStream(_input, "");
+		CharStream charStream(_input, "");
+
 		try
 		{
-			m_ast = yul::Parser(errorReporter, m_dialect).parse(_charStream);
-			if (!m_ast || !errorReporter.errors().empty())
+			ObjectParser parser(errorReporter, m_dialect);
+
+			auto scanner = make_shared<Scanner>(charStream);
+
+			if (!m_inputIsCodeBlock && scanner->currentToken() == Token::LBrace)
+				m_inputIsCodeBlock = true;
```

Comment:
Any reason this is not updated when `m_inputIsCodeBlock` is already true?
```suggestion
			m_inputIsCodeBlock = (scanner->currentToken() == Token::LBrace);
```

I don't think it makes much difference in the current implementation (it won't randomly flip from one to the other anyway) and just assigning it every time makes the logic clearer.